app/javascript/leaflet/controls/layer_opacity.js

```diff
@@ -9,6 +9,7 @@ export default class LayerOpacityControl extends Control {
     if (typeof layer.getLayers !== "undefined") {
       // Add first layer from layer group to options
       this.options.layer = layer.getLayers()[0];
+      this.options.layer = this.options.layer ? this.options.layer : layer;
```

Comment:
this looks like a case for the [nullish coalescing assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_assignment)!

```suggestion
      this.options.layer ??= layer;
```